policy-controller/k8s/index/src/pod.rs

```diff
@@ -68,6 +68,68 @@ pub(crate) fn tcp_port_names(spec: Option<k8s::PodSpec>) -> HashMap<String, Port
     port_names
 }
 
+/// Gets the container probe ports for a Pod.
+///
+/// The result is a mapping for each probe port exposed by a container in the
+/// Pod and the paths for which probes are expected. When a TCP probe is
+/// configured there is no path, so we use '/'.
+pub(crate) fn get_container_probes(
+    spec: &Option<k8s::PodSpec>,
+    port_names: &HashMap<String, PortSet>,
+) -> PortMap<HashSet<String>> {
+    let mut probes = PortMap::<HashSet<String>>::default();
+    for container in spec.iter().flat_map(|s| s.containers.iter()) {
+        set_probe_ports(&container.liveness_probe, &mut probes, port_names);
+        set_probe_ports(&container.readiness_probe, &mut probes, port_names);
+        set_probe_ports(&container.startup_probe, &mut probes, port_names);
+    }
+    probes
+}
+
+fn set_probe_ports(
+    probe: &Option<k8s::Probe>,
+    probes: &mut PortMap<HashSet<String>>,
+    port_names: &HashMap<String, PortSet>,
+) {
+    let probe = match probe {
+        Some(probe) => probe,
+        None => return,
+    };
+    if let Some(http) = &probe.http_get {
+        if let Some(path) = &http.path {
+            match &http.port {
```

Comment:
TIOLI.  this might be a bit easier to read as something like:

```
let ports = match &http.port {
    k8s::IntOrString::Int(port) => u16::try_from(*port).and_then(NonZeroU16::try_from).into_iter(),
    k8s::IntOrString::String(port) => port_names.get(port).into_iter().flatten(),
};
for port in ports {
let paths = probes.entry(port).or_default();
paths.insert(path.clone());
}
```

Reply:
(in particular, `p` tripped me up here because `path`, `probe`, and `port` all start with p)

Reply:
it occurs to me that if we're going to do it @adleong's way, this could even just return an iterator that's chained together in `get_container_probes` and `collect`ed...we don't *need* to write it that way, and i'm not sure if it's significantly more readable, but it's just a thought.